addons/website_sale/static/src/js/website_sale.js

```diff
@@ -193,6 +193,7 @@ $('.oe_website_sale').each(function () {
             $product_id.val(0);
             $parent.find(".js_check_product").attr("disabled", "disabled");
         }
+        $product_id.trigger('change');
```

Comment:
yes , we need to trigger it , because when end user change the quantity we have to check that product is in stock or not based on that we display him/her a warning message.